thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java

```diff
@@ -133,34 +136,42 @@ public void start() throws Exception {
         if (config.isPinotProxy()) {
           environment.jersey().register(new PinotDataSourceResource());
         }
+        if (config.isDetectionOnboard()) {
+          detectionOnboardServiceExecutor = new DetectionOnboardServiceExecutor();
+          detectionOnboardServiceExecutor.start();
+          environment.jersey().register(new DetectionOnboardResource(detectionOnboardServiceExecutor));
+        }
       }
 
       @Override
       public void stop() throws Exception {
-        if (config.isWorker()) {
+        if (config.isWorker() && taskDriver != null) {
```

Comment:
the first part is redundant. for cleanup use only "taskDriver != null"

Reply:
Good observation.